packages/wallet/ui/src/components/ConnectionSettingsDialog.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from 'react';
+import { makeStyles } from '@mui/styles';
+import Autocomplete from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
+import FormControl from '@mui/material/FormControl';
+import Button from '@mui/material/Button';
+import DialogTitle from '@mui/material/DialogTitle';
+import Dialog from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+// import DialogContentText from '@mui/material/DialogContentText';
+
+import DialogActions from '@mui/material/DialogActions';
+import { withApplicationContext } from '../contexts/Application';
+
+const useStyles = makeStyles(_ => ({
+  centeredText: {
+    textAlign: 'center',
+  },
+  dialog: {
+    minWidth: 240,
+  },
+}));
+
+const ConnectionSettingsDialog = ({
+  onClose,
+  open,
+  disconnect,
+  walletConnection,
+  setWalletConnection,
+  allWalletConnections,
+  setAllWalletConnections,
+}) => {
+  const classes = useStyles();
+  const [connection, setConnection] = useState(walletConnection);
+
+  const saveAndClose = () => {
+    if (connection) {
+      setWalletConnection(connection);
+      disconnect();
+      if (
+        !allWalletConnections.some(
+          c => c.label === connection.label && c.url === connection.url,
+        )
+      ) {
+        setAllWalletConnections(conns => [connection, ...conns]);
+      }
+    }
+    onClose();
+  };
+
+  return (
+    <Dialog onClose={onClose} open={open}>
+      <DialogTitle className={classes.centeredText}>
+        Connection Settings
+      </DialogTitle>
+      <DialogContent>
+        <FormControl sx={{ m: 1, minWidth: 120 }}>
+          <Autocomplete
+            value={connection}
+            id="connection"
+            options={allWalletConnections}
+            sx={{ width: 300 }}
+            onChange={(event, newValue) => {
+              if (typeof newValue === 'string') {
+                setConnection({
+                  url: newValue,
+                  label: newValue,
+                });
+              } else if (newValue && newValue.inputValue) {
+                setConnection({
+                  url: newValue.inputValue,
+                  label: newValue.inputValue,
+                });
+              } else {
+                setConnection(newValue);
+              }
+            }}
```

Comment:
this transformation seems worth a helper.
```suggestion
            onChange={(_, newValue) => setConnection(connectionValue(newValue))}
```